go/consensus/tendermint/abci/mux.go

```diff
@@ -422,6 +413,38 @@ func (mux *abciMux) InitChain(req types.RequestInitChain) types.ResponseInitChai
 
 	mux.logger.Debug("Genesis hook dispatch complete")
 
+	mux.RUnlock()
+
+	// TODO Matevz: It seems stateKeyGenesisRequest is still used in roothash's ForeignExecuteTx for generating new runtimes?
```

Comment:
Looked at that and it seems wrong to me as in `ForeignExecuteTx` it is using the roothash genesis block to populate the genesis block of a runtime that was just registered.

The runtime structure already contains a `Genesis` field that has either a `StateRoot` or `State`. I think we should:
* Remove `Blocks` from roothash genesis.
* Add `Round` to `RuntimeGenesis` and use that when initializing a runtime genesis block.
* Change `onNewRuntime` and `ForeignExecuteTx` to only use the runtime's `Genesis` structure.
* Update runtime-ethereum's genesis utilities to generate runtime descriptors instead of roothash state.

However this is independent of `InitChain` and so should be done in a separate issue/PR. I've filed #2426, maybe just reference it in this comment, e.g. `(see oasis-core#2426)`.